src/plugins/data-card/data-card-tile.tsx

```diff
@@ -53,8 +54,23 @@ export const DataCardToolComponent: React.FC<ITileProps> = observer((props) => {
         return model.title;
       }
     });
+    adjustHeight();
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
+  const adjustHeight = useCallback(() => {
+    if (!tileElt) return;
+    if (readOnly) onRequestRowHeight(model.id, height);
+    if (!readOnly){
+      const uiHeight = tileElt?.querySelector(".data-card-container")?.scrollHeight || 0;
+      const heightDiff = height ? height - uiHeight : 0;
+      heightDiff < 30 && onRequestRowHeight(model.id, uiHeight + 60);
+    }
+  }, [height, model.id, onRequestRowHeight, readOnly, tileElt]);
+
+  useEffect(() => {
+    adjustHeight();
```

Comment:
Did you try just putting the adjustHeight code directly in this useEffect? It would make the dependency list more clear and doesn't seem like it would make things any more complicated.

The calling of adjustHeight above in the first useEffect seems redundant with this one. They will both be called at the beginning when the component is first rendered. And then this one will be called again when it is re-rendered and the dependencies have changed.

The `currEditAttrId` and `currentEditFacet` dependencies are tricky here. There should be a comment about them. My understanding is that they are listed as dependencies so this will run as the user edits the card. Otherwise it wouldn't run when a new attribute was added.

Reply:
That worked - I think I put it in the loading effect at first for the sake of the left side.